module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,65 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+constexpr uint8_t ASCIIspace = 32;
+
+VectorPair compressGrayscale(const ArrayArray& data) {
+    VectorPair result;
+    result.reserve(width * height);
+
+    for (const auto& entry : data) {
+        uint8_t currValue = entry.front();
+        uint8_t currCount = 1;
+
+        for (uint8_t id = 1; id < width; id++) {
+            if (entry[id] == currValue) {
+                currCount++;
+            }
+            else {
+                result.emplace_back(std::make_pair(currValue, currCount));
+                currValue = entry[id];
+                currCount = 1;
+            }
+        }
+        result.emplace_back(std::make_pair(currValue, currCount));
```

Comment:
You can try to reconfigure for loop to use only one emplace_back

Reply:
+1